src/test/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistryTest.java

```diff
@@ -182,6 +183,23 @@ public void testGetArchiveRepoSpec() throws Exception {
         "  },",
         "  \"patch_strip\": 3",
         "}");
+    server.serve(
+        "/modules/baz/3.0/source.json",
+        """
+        {
+            "url": "https://example.com/archive.jar?with=query",
+            "integrity": "sha256-bleh",
+            "overlay": {
+                "BUILD.bazel": {
+                    "urls": [
+                        "http://mirror1",
```

Comment:
@Wyverald One thing I'm wondering about right now: There is no such thing for remote patches, but should we use the mirrors in `bazel_registry.json` to populate these URLs?

Reply:
I could see from the test is that mirrors is only used for the main URL of the module.
I also had the same thought that it was a bit confusing --

Since this is a distinct URL it actually doesn't even need to reside within BCR server so it probably needs it's own URLS.

in the simple/common case, it could be a relative file path and just use a single URL.

Maybe a follow up: if the URL is a `file://` use the bazel_registry.json mirrors?

I say let's finish the complete end-to-end with BCR first and then do a UX fixup once we test it out.

Reply:
taking a more serious look now -- shouldn't the overlay files be served from the registry itself? Just like how today patch files are served from the registry, instead of some random other URL. The only thing we're fetching from arbitrary URLs is the source archive itself.

I'm thinking that the `overlay` property can just be a JSON object (i.e. a dict) that maps from overlay file path to integrity (just like how `patches` is a map from patch file name to integrity). The overlay files will reside in the `overlay` directory next to the `patches` directory. So if you have the source.json:

```json
{
    "url": "...",
    "integrity": "...",
    "overlay": {
        "BUILD.bazel": "sha256-...",
        "mypkg/BUILD.bazel": "sha256-..."
    }
}
```

then your module version directory tree in the registry should look like:

```
|-- MODULE.bazel
|-- source.json
|-- overlay
    |-- BUILD.bazel
    |-- mypkg
        |-- BUILD.bazel
```

WDYT?

Reply:
I think I see this and construct the URL similar to how it's done for patches?